source_code/main_mcu/src/LOGIC/logic_gui.c

```diff
@@ -257,11 +257,41 @@ void logic_gui_erase_user(void)
     logic_smartcard_handle_removed();
 }
 
-/*! \fn     logic_gui_display_login_password(child_cred_node_t* child_node)
-*   \brief  Display login and password on the screen
+/*! \fn     logic_gui_create_TOTP_display_str(cust_char_t *TOTP_str, uint8_t TOTP_str_len, uint8_t remaining_secs)
+*   \brief  Create the TOTP display string: "<TOTP> [remaining secs]", example: "123456 [30]"
+*   \param  TOTP_str       String containing the TOTP
+*   \paran  TOTP_str_len   Total length of the TOTP_str
+*   \param  numDigits      Number of digits in TOTP
+*   \param  remaining_secs Number of seconds remaining
+*/
+static void logic_gui_create_TOTP_display_str(cust_char_t *TOTP_str, uint8_t TOTP_str_len, uint8_t numDigits, uint8_t remaining_secs)
```

Comment:
you can go uint16_t for the parameters :). 
you may also strip the static as we really don't use it much on the solution in general :)

Reply:
:+1:  on the uint6_t

Don't agree on the "static" but if that's what you want its your call :-). I just think it is good design/programming practise to limit scope. So much easier to read code and debug etc. When a function is static you immediately know that this function is only used in this translation unit.

Reply:
right, keep it static then :)